WalletWasabi/BranchNBound/BranchAndBound.cs

```diff
@@ -0,0 +1,178 @@
+using NBitcoin;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.BranchNBound
```

Comment:
Not sure if this should be here or where the other  coin selection components are. IMO it is not good to add a folder for this.

Reply:
I don't know about other such components. 

@adamPetho Do you possibly know?

Reply:
Previously we had different components and algos, so it made sense to make a folder for them. But that's not the case anymore, so yeah, the folder is not necessary at all.

Reply:
I have moved the code there.